doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
 
-Within each REGION there may be multiple CLUSTERs, which correspond roughly to timely dataflow instances.
-A CLUSTER contains indexes and is the unit within which it is possible to share indexes.
+Terms in this document that represent key user-facing concepts are introduced
+with *italics*. The underlying components may use different terminology
+internally, but any user-facing interfaces should use the term specified in this
+document. For example, the COMPUTE layer is free to use the term COMPUTE
+INSTANCE internally, but the user-facing documentation, web UI, and SQL API
+should all consistently use the term *cluster*.
 
-Within each REGION there may be multiple TIMELINEs, which correspond roughly to coordinator timelines.
-A TIMELINE contains collections, and is the unit within which it is possible to query across multiple collections
+## Overview
 
-The main restrictions we plan to impose on users is that their access to data is always associated with a REGION, CLUSTER, and TIMELINE, and attempts to access data across these may result in poorer performance, or explicit query rejection.
+Depicted below are the tangible concepts that Materialize Platform exposes to
+users.
 
-There is some work to do to present these concepts to users, but prior work exists with e.g. SQL `database`s for TIMELINE and Snowflake's `USE WAREHOUSE` command to pilot CLUSTER.
+![Diagram of user-facing concepts](./assets/user-concepts.png)
+
+An [*account*](#account) represents a customer of Materialize Platform. An
+account has [*users*](#user) and manages [*billing*](#billing).
+
+A [*region*](#region) is a specific region of a specific cloud provider (e.g.,
+aws:us-east-1). The region boundary is strong: besides users and billing,
+nothing is shared between regions in the same account.
+
+A [*cluster*](#cluster) is the user-facing name for a [COMPUTE
+INSTANCE](architecture-db.md#COMPUTE). A region can contain any number of clusters.
+
+A region contains a standard SQL catalog, which contains
+[*databases*](#database), which each contain [*schemas*](#schema), which each
+contain *database objects*. A database object is one of the following:
+
+  * a [*source*](#source), which imports data into the region;
+  * a [*sink*](#sink), which exports data out of the region;
+  * a [*table*](#table), which stores data directly in the region;
+  * a [*view*](#view), which describes a transformation of sources, tables, and
+    other views;
+  * an [*index*](#index), which schedules the persistent computation of a source
+    or view in a cluster;
+  * a [*secret*](#secret), which securely stores credentials for external
+    systems.
+
+Note that sources, tables, views, and secrets are shared across all clusters in
+the region, while indexes are only shared within a cluster.
+
+A [*timeline*](#timeline) represents a collection of sources (and views derived
+from those sources) that are in the same time domain. Timelines live somewhere
```

Comment:
"time domain" has an existing different meaning in coord.rs. Maybe "contemporaneous":

> A timeline represents a collection of sources (and views derived from those sources) that are contemporaneous.

Reply:
Oh? What's the difference between a "timeline" and a "time domain"? Or is "time domain" just what coord currently calls a time domain?

Reply:
timeline is the thing described by the timeline design doc (what unit the timestamp has). timedomain is "things we'd like to dump into a transaction that a user might want to read from".

Reply:
I don't care if this changes, but wanted to point out the existing definition.